### PR TITLE
Changes to compile on Windows

### PR DESCRIPTION
**Pull request by elikosan**
With these fixes i can install torch7 with luarocks on Windows.
Compiler: Visual Studio 2015
OS: Windows 7 Pro

### CODE REVIEW

File: lib/THNN/generic/BatchNormalization.c [hunk 1]
```diff
@@ -10,10 +10,10 @@ void THNN_(BatchNormalization_updateOutput)(
   bool train, double momentum, double eps)
 {
   long nInput = THTensor_(size)(input, 1);
-  long n = THTensor_(nElement)(input) / nInput;
+  long f,n = THTensor_(nElement)(input) / nInput;
 
   #pragma omp parallel for
-  for (long f = 0; f < nInput; ++f) {
+  for (f = 0; f < nInput; ++f) {
```
Comment (soumith) [inline lib/THNN/generic/BatchNormalization.c]: if you make this change, then you'll have to change the pragma to: #pragma omp parallel for private(f)

Comment (elikosan) [inline lib/THNN/generic/BatchNormalization.c]: Hi Soumith,

The loop iteration variables are private within their loops. So i believe it is not necessary to add private() here.

Eric.

**Comment by soumith** [comment]
Thanks!

**Comment by elikosan** [comment]
Thanks for the quick merge!
Best,
Eric.
